### DISCUSSION TITLE
Docker image dependencies

### DISCUSSION THREAD

**Discussion by nightly-raven** [knative/func#2409] [Q&A] [answered]
I know that functions are supposed to be built with minimal set-up overhead, but is there a way to install ffmpeg for example if my Python func depends on it? I haven't seen any Dockerfile related documentation.

**Comment by lkingland** [comment, marked as answer]
We do not currently have a way to specify a base image for a Function directly, however that is on the roadmap for the release after next.

For now, you can accomplish this by creating a custom Buildpack which has a Dockerfile for the "run" container with ffmpeg.   You can then specify this buildpack be applied in your `func.yaml`:
```
buildpacks:
  - [my buildpack]
```

This feature is a bit complex, and intended to allow for the extensibility of the system to entirely new language runtimes, so check back to see if we've implemented `--base-image=[my-image-with-ffmpeg]`  first!

**Reply by nightly-raven** [reply]
Thank you for following up on this matter!
